### PR TITLE
Bootstrap nimble

### PR DESCRIPTION
**Pull request by yyoncho**
PR split into 2 commits to make the reviewing easier.

Here it is the code that I used for generating the nimble bootstrap files

``` bash

#!/usr/bin/env bash

rm ~/.cache/nim/nimble_d/ -rf
nim cc --genScript src/nimble.nim
sed -i 's#-I.*nimble/src##' ~/.cache/nim/nimble_d/compile_nimble.sh
cp ~/Sources/nim/Nim/lib/nimbase.h ~/.cache/nim/nimble_d/
mkdir ~/.cache/nim/nim-workspace/ -p
rm -rf ~/Sources/nim/nim-workspace/scripts/nimble-linux/
cp -r ~/.cache/nim/nimble_d/ ~/Sources/nim/nim-workspace/scripts/nimble-linux/

# windows

rm ~/.cache/nim/nimble_d/ -rf
nim cc --os:windows --genScript src/nimble.nim
sed -i 's#-I.*nimble/src##' ~/.cache/nim/nimble_d/compile_nimble.bat
cp ~/Sources/nim/Nim/lib/nimbase.h ~/.cache/nim/nimble_d/
mkdir ~/.cache/nim/nim-workspace/ -p
rm -rf ~/Sources/nim/nim-workspace/scripts/nimble-windows/
cp -r ~/.cache/nim/nimble_d/ ~/Sources/nim/nim-workspace/scripts/nimble-windows/


```

### CODE REVIEW

**Comment by zah** [comment]
The Nimble C sources should be kept in a separate repo. This would allow us to easily prune the history of this repo to control the initial git clone size.

File: scripts/rebuild-nimble [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+if [ ! -f $NIMBLE ]; then
+    mkdir -p $NIMBLE_CACHE_DIR
+    pushd ./scripts/nimble-$OS/
+    $BUILD_COMMAND
+    popd
+    cp ./scripts/nimble-$OS/nimble $NIMBLE
+    mkdir bin -p
+    rm ./bin/nimble$EXE_SUFFIX
+    ln -s $NIMBLE ./bin/nimble$EXE_SUFFIX
```
Comment (zah) [inline scripts/rebuild-nimble:27]: At the moment, this script won't be able to switch back to a previously used Nimble version (e.g. for example, we may discover an issue in a Nimble upgrade and then decide to rollback to the previous version).

The solution may be another if that checks whether the link is up to date (or creating the link unconditionally outside of this if)

Comment (yyoncho) [reply scripts/rebuild-nimble:27]: I guess creating the link unconditionally should be good enough.

File: scripts/rebuild-nimble [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+if [ ! -f $NIMBLE ]; then
+    git submodule update --init
+    mkdir -p $NIMBLE_CACHE_DIR
+    pushd $NIMBLE_SOURCES
+    $BUILD_COMMAND
+    popd
+    cp $NIMBLE_SOURCES/nimble $NIMBLE
+fi
+
+mkdir bin -p
```
Comment (zah) [inline scripts/rebuild-nimble:28]:
```suggestion
mkdir -p bin
```

**Pull request merged**